sphinxcontrib/jupyter/writers/make_pdf.py

```diff
@@ -267,9 +268,13 @@ def process_tex_for_book(self, builder):
                     output.write(data)
                     f.close()
                     output.close()
-
-        self.nbconvert_index(builder)
+                    
         os.chdir(self.texbookdir)
+        self.nbconvert_index(builder)
         fl_tex = self.texbookdir + "/" + self.index_book + ".tex"
         filename = self.index_book
+
+        ## checking if an explicit output filename is specified in the config file
+        if "jupyter_pdf_book_name" in builder.config and builder.config["jupyter_pdf_book_name"]:
```

Comment:
@AakashGfude I suspect this needs to check `jupyter_pdf_book` as the first condition above will always be met even if it is `None` (i.e. as the default value)

Reply:
@AakashGfude can you check this condition?